pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -0,0 +1,109 @@
+"""Plugin for constructor definitions."""
+import astroid
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+class HassConstructorFormatChecker(BaseChecker):  # type: ignore[misc]
+    """Checker for setup definitions."""
+
+    __implements__ = IAstroidChecker
+
+    name = "hass_enforce_type_hints"
+    priority = -1
+    msgs = {
+        "W0012": (
```

Comment:
Let's keep few numbers for previous plugins and and start from `W0016`.

Reply:
Is W the right code for this? Or should we use one of the other codes?
> There are multiple message categories, these being C, W, E, F, R, standing for Convention, Warning, Error, Fatal and Refactoring

Reply:
Well, for simplicity I think it's ok to use W for all custom checks.